packages/plugins/Filter/src/utils/updateFeatureVisibility.ts

```diff
@@ -0,0 +1,148 @@
+import { Feature, Map } from 'ol'
+import Style from 'ol/style/Style'
+import { FilterConfiguration } from '@polar/lib-custom-types'
+import { DatePattern, FilterState, LayerId, TimeOption } from '../types'
+
+// comparable style to identify elements *supposed* to be invisible
+export const Invisible = new Style()
+
+const doesFeaturePassCategoryFilter = (
+  categories: FilterConfiguration['layers'][string]['categories'],
+  category: FilterState['category'],
+  feature: Feature,
+  layerId: LayerId
+) =>
+  (categories || []).every(
+    ({ targetProperty }) =>
+      category[layerId][targetProperty][feature.get(targetProperty)]
```

Comment:
Does this make sure that the targetProperty exists on the feature? Or what is done here?

Reply:
There may be multiple categories a feature is filtered by. This checks if the checkbox value is `true` on all category filters. A feature filtered by one category isn't shown at all.